### DISCUSSION TITLE
Error while sending data

### DISCUSSION THREAD

**Discussion by Pitu** [savonet/liquidsoap#2915] [User Support]
I'd like to preface this question by saying that I'm using a deprecated version of liquidsoap and planning to upgrade this week to the stable version.
We're running into a few issues ever since we had to provision new servers and move providers, and we can't seem to pinpoint the origin of the problem we are having.

At the moment we are running Liquidsoap 1.4.3 with a script that is fairly simple in nature, which I'll be attaching below. The problem happens pretty frequently, and the error log shows the following:

```
2023/02/13 01:32:04 [clock.wallclock_main:2] We must catchup 14.36 seconds!
2023/02/13 01:32:04 [icecast:2] Error while sending data: could not write data to host: Broken pipe in write()!
2023/02/13 01:32:04 [icecast:3] Closing connection...
2023/02/13 01:32:04 [icecast:3] Will try to reconnect in 3.00 seconds.

2023/02/13 01:32:08 [icecast:3] Connecting mount /fallback for source@127.0.0.1...
2023/02/13 01:32:08 [icecast:3] Connection setup was successful.
2023/02/13 01:32:08 [icecast:3] Connecting mount /stream for source@127.0.0.1...
2023/02/13 01:32:08 [icecast:3] Connection setup was successful.
2023/02/13 01:32:08 [icecast:3] Connecting mount /opus for source@127.0.0.1...
2023/02/13 01:32:08 [icecast:3] Connection setup was successful.
```

Basically it starts by saying it needs to catchup, even though looking at the CPU usage during the entire period nothing weird pops up, it even stays below 20% so it's not maxing the available bandwidth for sure.
And after a few of those catchups it closes the connection with that broken pipe. So my question would be what could be causing that broken pipe in the first place? Is this an issue that can originate when waiting for the `js` script to print the full path of the file to be played or is it referring to the data from liquidsoap to icecast?

And this seems to be happening every 5 or minutes, generating a lot of stutters on the stream until it eventually disconnects altogether.

Thanks

```python
set("log.file.path", "/home/radio/radio.log")
set("log.stdout",true)
set("log.level",3)

enable_replaygain_metadata()

# Our custom request function
def get_request() =
  # Get the URI to the file to be played
  uri = list.hd(default="", get_process_lines("node /home/radio/getSong.js"))
  print("Uri: #{uri}")
  # Create a request
  request.create(uri)
end

s = request.dynamic(id="s", get_request)
s = amplify(1.,override="replay_gain",s)

output.icecast(%opus(
  channels=2,
  application="audio",
  samplerate=48000,
  bitrate=256,
  signal="music"),
  encoding="UTF-8",
  id="icecast",
  mount="/opus",
  format="audio/ogg",
  icy_metadata="false",
  name="----",
  host="127.0.0.1",
  port=7890,
  password="----",
  genre="----",
  url="----",
  description = "----",
  public=true,
  mksafe(s))

output.icecast(%vorbis.cbr(
  samplerate=48000,
  channels=2,
  bitrate=256),
  encoding="UTF-8",
  id="icecast",
  mount="/stream",
  name="----",
  host="127.0.0.1",
  port=7891,
  password="----",
  genre="----",
  url="----",
  description = "----",
  public=true,
  mksafe(s))

output.icecast(%mp3(
  samplerate=44100,
  stereo=true,
  bitrate=256),
  encoding="UTF-8",
  id="icecast",
  mount="/fallback",
  name="----",
  host="127.0.0.1",
  port=7892,
  password="----",
  genre="----",
  url="----",
  description = "----",
  public=true,
  mksafe(s))
```

**Comment by toots** [comment]
Hi! The source of delay appears to be coming from the network connection error. The socket is blocked trying to send data, which creates some delay in the streaming loop.
